src/octoprint/util/comm.py

```diff
@@ -387,7 +388,21 @@ def __init__(self, port = None, baudrate=None, callbackObject=None, printerProfi
 		self._currentLine = 1
 		self._line_mutex = threading.RLock()
 		self._resendDelta = None
-		self._lastLines = deque([], 50)
+
+		self._use_xonxoff_workaround = settings().get(["serial", "useSoftwareFlowWorkaround"])
+		self._xonxoff_stopChar = chr(19)
+		self._xonxoff_resumeChar = chr(17)
+		self._waitingForM28Confirmation = False
+		self._startedSendingFileToSD = False
```

Comment:
Can you explain what this flag is accomplishing? I understand it's your solution to the line number test, but I'd like to understand your reasoning here and figure out if there is a better approach. I'm not too happy about introducing so many one shot variables here for one feature, and this is the most puzzling of all.

Why is just using the state like your previously did not working here?

Also, if kept this should be changed to snake_case instead of camelCase. Yes, lot's of old vars in there still in camelCase, but I'm slowly migrating that and am also trying not to introduce new ones.

Reply:
At first I tried using the state, but it locked. I'll try to explain: the send_loop blocks until a clear to send it's sent, in handle_ok, if I ignore the ok handling using the state will not send a clear_to_send needed to unblock send_loop. The idea is that, the ok_handle should ignore handling the ok only after the confirmation of M28 is received. The state does not catch this... that's why I use another switch :/ (the  _waitingForM28Confirmation). Another problem is that the state also is off when the file is actually finished. When it's waiting for ok confirmation, the send queue, will be filled with 1 item (line). But when the workaround is active, the send queue will be filled with more lines, there could be  hundreds of lines in the queue and the state is changed when all the lines from the file are set in the queue, but not removed. So for big files, the progress bar will be 100% done, the popup will show the transfer is finished, and the state is changed, but the file is still transferring (what's left in the send queue). So I needed another switch to detect when the file is actually finished (and this is when M29 is 'actually' sent). And this switch _startedSendingFileToSD  is used to actually catch this and to filter the flow control, if I use the state, what's left in the queue will not be filtered properly and it could lead to issues (the printer sends a stop and it will be ignored). 

I hope I've explained it well and will help in the future when updating the comm layer :)